pmix_governance.tex

```diff
@@ -633,6 +635,30 @@ \subsubsection{Minor Document Changes Guidance}%
 the changes will be handled using the full rigorous process
 as described in Section~\ref{the-standardization-process}.
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\hypertarget{abi-changes-guidance}{%
+\subsubsection{Application Binary Interface (ABI) Changes Guidance}%
+\label{abi-changes-guidance}}
+
+An \acf{API} defines how data types, data structures, and functions are represented in source code.
+An \acf{ABI} defines how data types, data structures, and functions are represented in machine code for a given system.
+An important aspect of an \ac{ABI} is the size, layout, and alignment of data structures.
+A stable \ac{ABI} may allow a program compiled with one implementation of the PMIx Standard to run with a different implementation of the PMIx Standard as long as both implementations adhere to the same \ac{ABI}.
+
+The PMIx Standard should strive to maintain a stable \ac{ABI} to support applications and tools that rely on more than one implementation of the PMIx Standard.
+To facilitate such interoperability the PMIx \ac{ASC} will maintain, to the best of their ability, a standard set of headers that are versioned with the PMIx Standard that applications and tools can reference~\footnote{PMIx Headers for ABI Compatibility \url{https://github.com/pmix/pmix-abi}}.
+
```

Comment:
Add the following:
-----
The PMIx Standard ABI is defined in two parts. The Stable ABI represents the Stable PMIx Standard elements (see Section 1.3.1 of the PMIx Governance Document). The Provisional ABI represents the Provisional PMIx Standard elements (see Section 1.3.1 of the PMIx Governance Document). The rationale for this separation is in recognition that Provisional PMIx Standard elements are able to change frequently in response to broad experimentation. This is in contrast to Stable PMIx Standard elements which are permanent, non-changing parts of the PMIx Standard.

Both the Stable and Provisional ABIs are versioned with two increasing numbers (largely following the Semantic Versioning 2.0.0 specification [1]):
 * `MAJOR` incremented when the ABI changed in a backwards incompatable manner.
 * `MINOR` incremented when functionality is added to the ABI in a backwards compatible manner.

The PMIx ASC will maintain a mapping between the PMIx Standard version and the ABI version.

[1] https://semver.org/spec/v2.0.0.html

Reply:
Addressed in the newest commit.